Structured Kernels: Use at::detail::empty functions

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #71640
* #70618
* __->__ #70617

This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

Differential Revision: [D33623680](https://our.internmc.facebook.com/intern/diff/D33623680)

peterbell10 committed:
Structured Kernels: Use at::detail::empty functions

This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33623680).

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

Differential Revision: [D33623680](https://our.internmc.facebook.com/intern/diff/D33623680)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33623680).

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

Differential Revision: [D33623680](https://our.internmc.facebook.com/intern/diff/D33623680)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33623680).

peterbell10 committed:
Update on "Structured Kernels: Use at::detail::empty functions"


This reduces the divergence between the code generated for
`create_out` different devices, and means the `TensorOptions` don't
need to be unpacked.

Differential Revision: [D33623680](https://our.internmc.facebook.com/intern/diff/D33623680)

[ghstack-poisoned]

ngimel commented:
Linalg failures are unrelated, right? I'm landing this.
